src/ngMessageFormat/messageformat.js

```diff
@@ -0,0 +1,938 @@
+  this.index = match.index;
+  this.expressionFn = this.$parse(this.text.substring(this.expressionStartIndex, this.index));
+  if (goog.DEBUG) {
+    // Only needed in order to pretend to be $interpolate for tests copied from interpolateSpec.js
+    this.expressionFn.exp = this.text.substring(this.expressionStartIndex, this.index);
+  }
+  this.rule = null;
+};
+
+
+function $$MessageFormatProvider() {
+  this['$get'] = ['$parse', '$locale', '$sce', '$exceptionHandler', function $get(
+                   $parse,   $locale,   $sce,   $exceptionHandler) {
+
+    function getStringifier(trustedContext, allOrNothing, text) {
+      return function stringifier(value) {
+        try {
+          var value = trustedContext ? $sce['getTrusted'](trustedContext, value) : $sce['valueOf'](value);
+          return allOrNothing && (value === void 0) ? value : stringify(value);
+        } catch (err) {
+          var newErr = $$messageFormatMinErr('interr', "Can't interpolate: {0}\n{1}", text,
+            err.toString());
+          $exceptionHandler(newErr);
+        }
+      };
+    }
+
+    function parseMustache(text, mustHaveExpression, trustedContext, allOrNothing) {
+      var stringifier = getStringifier(trustedContext, allOrNothing, text);
+      var parser = new MessageFormatParser(text, 0, $parse, $locale['pluralCat'], stringifier,
+                                           mustHaveExpression, trustedContext, allOrNothing);
+      parser.run(parser.ruleMustache);
+      return parser.parsedFn;
+    }
+
+    function interpolate(text, mustHaveExpression, trustedContext, allOrNothing) {
+      var stringifier = getStringifier(trustedContext, allOrNothing, text);
+      var parser = new MessageFormatParser(text, 0, $parse, $locale['pluralCat'], stringifier,
+                                           mustHaveExpression, trustedContext, allOrNothing);
+      parser.run(parser.ruleInterpolate);
+      return parser.parsedFn;
+    }
+
+    function cacheCall(fn, text, mustHaveExpression, trustedContext, allOrNothing) {
+      // TODO: How should the cache work in the presence of mustHaveExpression, trustedContext and allOrNothing
+      //       This is broken until then.  So we just use the uncached call.
+      return fn(text, mustHaveExpression, trustedContext, allOrNothing);
+      /*
+       *var parsedFn = PARSE_CACHE_FOR_INTERPOLATIONS[text];
+       *if (parsedFn == null) {
+       *  parsedFn = PARSE_CACHE_FOR_INTERPOLATIONS[text] = fn(text, mustHaveExpression, trustedContext, allOrNothing);
+       *}
+       *return parsedFn;
+       */
+    }
+
+    return {
+      'interpolate': function cachedInterpolate(text, mustHaveExpression, trustedContext, allOrNothing) {
+        return cacheCall(interpolate, text, mustHaveExpression, trustedContext, allOrNothing);
+      },
+      'parseMustache': function cachedParseMustache(text, mustHaveExpression, trustedContext, allOrNothing) {
```

Comment:
Yes, it was for a bunch of intermediate commits before I would replace $interpolate completely.  I decided to replace $interpolate fully sooner than later so it's no longer required.

As a heads up, in a future commit, I plan to `$decorate` our original `$parse` to support these extensions to allow things like `ng-bind` to work well (as wells as Angular 2 as well.)  Luckily, the parser is architected such that one can parse a varied set of things just by picking the correct starting state and "enclosing" expressions such push onto the rule stack before dispatching to them so this shouldn't be big change to the parser.